web/src/message_list.ts

```diff
@@ -503,14 +499,16 @@ export class MessageList {
 
     remove_and_rerender(message_ids: number[]): void {
         const should_rerender = this.data.remove(message_ids);
+        this.update_user_sidebar_participants();
         if (!should_rerender) {
             return;
         }
         this.rerender();
-        // Rebuild message list if we're deleting messages from the current list,
-        // since we might need to remove a participant user.
+    }
```

Comment:
I think we need to do this here too, to avoid leaving a muted user visible.

```
diff --git a/web/src/message_list.ts b/web/src/message_list.ts
index 9c5cefe235..242830663b 100644
--- a/web/src/message_list.ts
+++ b/web/src/message_list.ts
@@ -614,6 +614,7 @@ export class MessageList {
         // But in any case, we need to rerender the list for user muting,
         // to make sure only the right messages are hidden.
         this.rerender();
+        this.update_user_sidebar_participants();
     }
 
     all_messages(): Message[] {
```